### PR TITLE
fix(store): overrideSelector allows for function overrides too

### LINKED ISSUE #2825: mockStore.overrideSelector(selector: Selector, value: Value) does not work
The docs for `overrideSelector()` allow for the selector to be passed in to be of the function form or a string.

```
  overrideSelector<Selector extends Memoized<Result>, Value extends Result, Result = Selector extends MemoizedSelector<any, infer T> ? T : Selector extends MemoizedSelectorWithProps<any, any, infer U> ? U : Value>(selector: string | Selector, value: Value): OnlyMemoized<typeof selector, Result>
```

MockStore's implementation of select however will only allow 'string' forms to be overridden;

```
  select(selector: any, prop?: any) {
    if (typeof selector === 'string' && this.selectors.has(selector)) {
      return new BehaviorSubject<any>(
        this.selectors.get(selector)
      ).asObservable();
    }

    return super.select(selector, prop);
  }
```

Removing the typecheck seemed sufficient in my 'works for me' testing.

### PR DESCRIPTION
**Pull request by bdirito**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?
fixes #2825 

```
[x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: modules/store/testing/src/mock_store.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class MockStore<T = object> extends Store<T> {
   }
 
   select(selector: any, prop?: any) {
-    if (typeof selector === 'string' && this.selectors.has(selector)) {
+    if (this.selectors.has(selector)) {
```
Comment (timdeschryver) [inline modules/store/testing/src/mock_store.ts:106]: Can you add a test for the changed behavior?
It's unclear for me which extra cases this change adds, that weren't supported previously.

Comment (bdirito) [reply modules/store/testing/src/mock_store.ts:106]: Such a case should have been covered by `modules/store/testing/spec/mock_store.spec.ts:162` (vs 152) in the test    `it('should allow mocking of store.select with a memoized selector using overrideSelector', () => {` however when i attempt to run this test as explained by the docs 
```
yarn && yarn jest modules/store/testing/spec/mock_store.spec.ts --watch
```

 i get.

```
ts-jest[versions] (WARN) Version 4.0.5 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=2.7.0 <4.0.0). Please do not report issues in ts-jest if you are using unsupported versions.
 FAIL  modules/store/testing/spec/mock_store.spec.ts
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    modules/store/testing/spec/mock_store.spec.ts:314:33 - error TS7006: Parameter 'done' implicitly has an 'any' type.

[...more...]
```

Comment (bdirito) [reply modules/store/testing/src/mock_store.ts:106]: Without digging too much into your build pipeline I'm wondering if these tests are getting run. The circle build for this pr 
https://app.circleci.com/pipelines/github/ngrx/platform/749/workflows/54e1e0a8-e3ed-4108-a4fa-fda9d5c5d636/jobs/31234

Includes 
```
#!/bin/bash -eo pipefail
yarn nx affected:test --base=$(cat ~/project/master.txt) --head=$CIRCLE_SHA1 --parallel
yarn run v1.17.3
$ nx affected:test --base=17571e5b --head=c4a3d5021f58bc73bc964620d061fe3b27225378 --parallel

>  NX  No projects with "test" were run


———————————————————————————————————————————————

>  NX   SUCCESS  Running target "test" succeeded


Done in 3.09s.
```

This looks offhand like no tests were actually run, that *with* the source code file edit to `modules/store/testing/src/mock_store.ts`

**Comment by brandonroberts** [comment]
I don't see how this change is needed, as mocking selectors this way is functioning currently as intended. The string check is to support string selectors where you don't have a `setResult` method to override the value.

**Pull request closed without merging**